agent/src/main/java/com/teamscale/jacoco/agent/testimpact/TestExecutionWriter.java

```diff
@@ -28,8 +33,8 @@ public void append(TestExecution testExecution) throws IOException {
 		if (!testExecutionFile.exists()) {
 			this.testExecutionFile.createNewFile();
 		}
-		FileWriter writer = new FileWriter(testExecutionFile);
-		GSON.toJson(testExecutionsList, writer);
-		writer.close();
+		try (BufferedSink writer = Okio.buffer(Okio.sink(testExecutionFile))) {
```

Comment:
Do we still want to call that "writer"? It does not look like it behaves like a typical Writer anymore.